lib/publiccloud/instance.pm

```diff
@@ -42,10 +44,51 @@ sub run_ssh_command {
 
     $args{timeout} //= SSH_TIMEOUT;
 
-    my $ssh_cmd = sprintf('ssh -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no -oLogLevel=ERROR -i "%s" "%s@%s" -- %s',
-        $self->ssh_key, $self->username, $self->public_ip, $args{cmd});
+    my $ssh_cmd = sprintf('ssh %s -i "%s" "%s@%s" -- %s',
+        $self->ssh_opts, $self->ssh_key, $self->username, $self->public_ip, $args{cmd});
     record_info('CMD', $ssh_cmd);
     return script_output($ssh_cmd, $args{timeout}, %args);
 }
 
+=head2 scp
+
+    scp($from, $to, timeout => 90);
+
+Use scp to copy a file from or to this instance. A url starting with
+C<remote://> is replaces with the IP from this instance. E.g. a call to copy
+the file I</var/log/cloudregister> to I</tmp> looks like:
+C<<<$instance->scp('remote://var/log/cloudregister', '/tmp');>>>
+=cut
+sub scp {
+    my ($self, $from, $to, %args) = @_;
+
+    my $url = sprintf('%s@%s://', $self->username, $self->public_ip);
+    $from =~ s/^remote:\/\//$url/;
+    $to   =~ s/^remote:\/\//$url/;
+
+    my $ssh_cmd = sprintf('scp %s -i "%s" "%s" "%s"',
+        $self->ssh_opts, $self->ssh_key, $from, $to);
+
+    return script_run($ssh_cmd, $args{timeout});
+}
+
+=head2 upload_log
+
+    upload_log($filename);
+
+Upload a file from this instance to openqa using L<upload_logs()>.
+If the file doesn't exists on the instance, B<no> error is thrown.
+=cut
+sub upload_log {
+    my ($self, $remote) = @_;
+
+    my $tmpdir = script_output('mktemp -d');
+    my $dest   = $tmpdir . '/' . basename($remote);
+    my $ret    = $self->scp('remote://' . $remote, $dest);
+    if (defined($ret) && $ret == 0) {
+        upload_logs($dest, failok => 1);
+    }
+    assert_script_run('rm -rf ' . $tmpdir);
```

Comment:
are you sure that you want to fail in case failure to delete temp dir ?

Reply:
with flag `-r` will always return 0

Reply:
> with flag `-r` will always return 0

ah in this case for sure we should go with 
```suggestion
    script_run('rm -rf ' . $tmpdir);
```

Reply:
Oh didn't knew that `rm -rf` always return 0.
But in this case It doesn't metter or?

What do you think of: 
```suggestion
    assert_script_run('test -d ' . $tmpdir . ' && rm -rf ' . $tmpdir);
```

Reply:
Why not keeping it simple with script_run/assert_script_run, as it always returns 0? They way it is now will never fail...

Reply:
> Oh didn't knew that `rm -rf` always return 0.
> But in this case It doesn't metter or?

for me it is still matters , probably even more. It is about code readability . when I see `assert_script_run` I am doing some conclusions during test failure investigation. So such a "trap"  could dramatically increase time for investigation what actually fail.

But before going forward with discussion I want to understand if we really want to fail when not able to delete temp folder and why it is so important ?

Reply:
Because this is a indication that something fail with creating the tempdirectory or get the name of the tmpdirectory.

Reply:
why you are running `upload_logs` with `fail_ok` then ?

Reply:
Good catch, this doesn't make sense, as I previously check if the scp is ok.